src/store/poll-page/usePollStore.ts

```diff
@@ -0,0 +1,65 @@
+import { create } from 'zustand';
+
+import { SendingStatus } from '@/components/pages/poll-page/components/poll-form/types';
+import { Answer, QuestionDisplay, QuestionType } from '@/types/poll';
+import { Category } from '@/types/poll';
+type Action = {
+  setIsValid: (newValue: boolean) => void;
+  setQuestionsListOpened: (newValue: boolean) => void;
+  setCurrentQuestions: (newValue: Category) => void;
+  setAnswers: (newAnswers: Answer[]) => void;
+  setCurrentCategory: (newCategory: number) => void;
+  setIsSendingStatus: (newStatus: SendingStatus) => void;
+  reset: () => void;
+};
+
+type State = {
+  isValid: boolean;
+  isQuestionsListOpened: boolean;
+  currentQuestions: Category;
+  answers: Answer[];
+  currentCategory: number;
+  sendingStatus: SendingStatus;
+};
+
+const initialValue: State = {
+  isQuestionsListOpened: false,
+  isValid: false,
+  currentQuestions: {
+    name: '',
+    count: 1,
+    questions: [
+      {
+        id: '',
+        name: '',
+        criteria: '',
+        isRequired: true,
+        text: '',
+        type: QuestionType.TOGGLE,
+        description: '',
+        display: QuestionDisplay.AMOUNT,
+      },
+    ],
+  },
+  answers: [],
+  currentCategory: 0,
+  sendingStatus: SendingStatus.ANY,
+};
+
+export const usePollStore = create<State & Action>()(set => ({
+  ...initialValue,
+
+  setQuestionsListOpened: (newValue: boolean) =>
+    set({ isQuestionsListOpened: newValue }),
+  setIsValid: (newValue: boolean) => set({ isValid: newValue }),
+  setCurrentQuestions: (newValue: Category) =>
+    set({ currentQuestions: newValue }),
+  setAnswers: (newAnswers: Answer[]) => set({ answers: newAnswers }),
+  setCurrentCategory: (newCategory: number) =>
+    set({ currentCategory: newCategory }),
+  setIsSendingStatus: (newStatus: SendingStatus) =>
+    set({ sendingStatus: newStatus }),
```

Comment:
I don't think you need this setters, updating values in zustand could be done using setState to update the store

And overall using zustand might be an overkill here, u could get away using vanilla react Context

Reply:
simply replace setters with `setState` in places where u use setters and remove setters, no need to remove `zustand`